tests/indices/list/test_base.py

```diff
@@ -187,11 +209,58 @@ def test_query(
     assert node_info["end"] == 12
 
 
+def mock_text_splitter(
+    prompt: Prompt, num_chunks: int, padding: Optional[int] = 1
+) -> TokenTextSplitter:
+    """Mock text splitter."""
+    return TokenTextSplitter(
+        separator=" ",
+        chunk_size=30,
+        chunk_overlap=10,
+        tokenizer=globals_helper.tokenizer,
+    )
+
+
+@patch.object(LLMPredictor, "total_tokens_used", return_value=0)
+@patch.object(LLMPredictor, "predict", side_effect=mock_llmpredictor_predict)
+@patch.object(LLMPredictor, "__init__", return_value=None)
+@patch.object(
+    PromptHelper, "get_text_splitter_given_prompt", side_effect=mock_text_splitter
+)
+def test_index_overlap(
+    _mock_token_text_splitter: Any,
+    _mock_init: Any,
+    _mock_predict: Any,
+    _mock_total_tokens_used: Any,
+    documents_with_overlap: List[Document],
+    struct_kwargs: Dict,
+) -> None:
+    """Test node info calculation with overlapping node text."""
+    index_kwargs, query_kwargs = struct_kwargs
+    index = GPTListIndex(documents_with_overlap, **index_kwargs)
+
+    query_str = "What is?"
+    response = index.query(query_str, mode="default", **query_kwargs)
+    print(response.source_nodes)
+    node_info_0 = (
+        response.source_nodes[0].node_info if response.source_nodes[0].node_info else {}
+    )
+    assert node_info_0["start"] == 0
```

Comment:
for readability can we add comments as to what the numbers represent (e.g. "index corresponding to the end of text \"this is a test")